### PR TITLE
Fill in _posts/2021-03-17-#21142.md

### PR DESCRIPTION
**Pull request by MarcoFalke**

### CODE REVIEW

File: _posts/2021-03-17-#21142.md [hunk 1]
```diff
@@ -10,13 +10,51 @@ status: upcoming
...
+
+- `MemPoolAccept` is used to accept (or reject) transaction from remote peers, the wallet and the RPC interface.
+
+- The existing `process_message` fuzz target can also test messages of type
+  `tx`. It works by [initializing a
+  blockchain](https://github.com/bitcoin/bitcoin/blob/63314b8211d795b2bf0814e244d801e74f50b152/src/test/fuzz/process_message.cpp#L63),
+  then it lets the fuzz engine create a message (e.g. `tx`) from a peer [to be
+  processed](https://github.com/bitcoin/bitcoin/blob/63314b8211d795b2bf0814e244d801e74f50b152/src/test/fuzz/process_message.cpp#L95).
+
+- The new `tx_pool` fuzz target aims to fuzz mempool acceptance more efficient.
```
Comment (jonatack) [inline _posts/2021-03-17-#21142.md:36]:
```suggestion
- The new `tx_pool` fuzz target aims to make fuzz mempool acceptance more efficient.
```

File: _posts/2021-03-17-#21142.md [hunk 2]
```diff
@@ -10,13 +10,51 @@ status: upcoming
...
+1.  Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+    NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+    What was your review approach?
+
+2. Why does the existing `process_message_tx` fuzz target perform so poorly
+   when it comes to mempool acceptance?
+
+3. Why does the newly added [`tx_pool` fuzz target](https://github.com/bitcoin/bitcoin/blob/e4e253d73007e0b680d2a473327c6fd66de4d86c/src/test/fuzz/tx_pool.cpp#L203) achieve higher coverage in `MemPoolAccept` than the `process_message_tx` target?
+
+4. Is it expected to see more transactions rejected or accepted to the mempool in the `tx_pool` target? Why? You may collect evidence to support your answer by debugging the `tx_pool` target or by assuming all fields in the `ConsumeTransaction` helper are initialized by values picked uniformly random (Real fuzz engines do not pick values uniformly random, but this assumption is good enough for now).
```
Comment (jonatack) [inline _posts/2021-03-17-#21142.md:53]:
```suggestion
4. Is it expected to see more transactions rejected or accepted to the mempool in the `tx_pool` target? Why? You may collect evidence to support your answer by debugging the `tx_pool` target or by assuming all fields in the `ConsumeTransaction` helper are initialized by values picked to be uniformly random (Real fuzz engines do not pick values uniformly randomly, but this assumption is good enough for now).
```

File: _posts/2021-03-17-#21142.md [hunk 3]
```diff
@@ -10,13 +10,51 @@ status: upcoming
...
+
+5. How does the `tx_pool_standard` fuzz target improve upon the `tx_pool` fuzz target even further?
```
Comment (jonatack) [inline _posts/2021-03-17-#21142.md:55]:
```suggestion
5. How does the `tx_pool_standard` fuzz target improve even further on the `tx_pool` fuzz target?
```

**Review by jonatack — COMMENTED** [review]
Thanks @MarcoFalke, looks great. Good idea to add the fuzz trophies wiki. A couple quick comments.

File: _posts/2021-03-17-#21142.md (hunk 3 above)
Comment (jonatack) [reply _posts/2021-03-17-#21142.md:55]: (or just s/upon/on/)

File: _posts/2021-03-17-#21142.md (hunk 1 above)
Comment (MarcoFalke) [reply _posts/2021-03-17-#21142.md:36]: I think this is right `fuzz` is the verb, not a noun

File: _posts/2021-03-17-#21142.md (hunk 2 above)
Comment (jonatack) [inline _posts/2021-03-17-#21142.md:53]:
```suggestion
4. Is it expected to see more transactions rejected or accepted to the mempool in the `tx_pool` target? Why? You may collect evidence to support your answer by debugging the `tx_pool` target or by assuming all fields in the `ConsumeTransaction` helper are initialized by uniformly random values (Real fuzz engines do not pick uniformly random values, but this assumption is good enough for now).
```

Comment (jonatack) [reply _posts/2021-03-17-#21142.md:53]: Second try on the suggestion

File: _posts/2021-03-17-#21142.md (hunk 1 above)
Comment (jonatack) [reply _posts/2021-03-17-#21142.md:36]: Ah, then s/efficient/efficiently/

Comment (MarcoFalke) [inline _posts/2021-03-17-#21142.md:36]:
```suggestion
- The new `tx_pool` fuzz target aims to fuzz mempool acceptance more efficiently.
```

Comment (MarcoFalke) [reply _posts/2021-03-17-#21142.md]: thanks, done

File: _posts/2021-03-17-#21142.md (hunk 2 above)
Comment (MarcoFalke) [reply _posts/2021-03-17-#21142.md]: picked

File: _posts/2021-03-17-#21142.md [hunk 4]
```diff
@@ -10,13 +10,51 @@ status: upcoming
...
+1.  Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+    NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+    What was your review approach?
+
+2. Why does the existing `process_message_tx` fuzz target perform so poorly
+   when it comes to mempool acceptance?
+
+3. Why does the newly added [`tx_pool` fuzz target](https://github.com/bitcoin/bitcoin/blob/e4e253d73007e0b680d2a473327c6fd66de4d86c/src/test/fuzz/tx_pool.cpp#L203) achieve higher coverage in `MemPoolAccept` than the `process_message_tx` target?
+
+4. Is it expected to see more transactions rejected or accepted to the mempool in the `tx_pool` target? Why? You may collect evidence to support your answer by debugging the `tx_pool` target or by assuming all fields in the `ConsumeTransaction` helper are initialized by values picked to be uniformly random (Real fuzz engines do not pick values uniformly randomly, but this assumption is good enough for now).
```
Comment (jonatack) [inline _posts/2021-03-17-#21142.md:53]: oops didn't see before: s/Real/real/

Comment (MarcoFalke) [inline _posts/2021-03-17-#21142.md:53]:
```suggestion
4. Is it expected to see more transactions rejected or accepted to the mempool in the `tx_pool` target? Why? You may collect evidence to support your answer by debugging the `tx_pool` target or by assuming all fields in the `ConsumeTransaction` helper are initialized by values picked to be uniformly random. Real fuzz engines do not pick values uniformly randomly, but this assumption is good enough for now.
```

File: _posts/2021-03-17-#21142.md [hunk 5]
```diff
@@ -10,13 +10,51 @@ status: upcoming
...
+
+- Bitcoin Core has [over
+  160](https://github.com/bitcoin/bitcoin/tree/master/src/test/fuzz) individual
+  fuzz targets, but the code coverage of consensus, validation and P2P code
+  paths is [still
+  low](https://marcofalke.github.io/btc_cov/fuzz.coverage/index.html). For
+  example `MemPoolAccept::AcceptSingleTransaction` never gets past `PreChecks`
+  and exits early.
+
+- `MemPoolAccept` is used to accept (or reject) transaction from remote peers, the wallet and the RPC interface.
```
Comment (jonatack) [inline _posts/2021-03-17-#21142.md:28]:
```suggestion
- `MemPoolAccept` is used to accept (or reject) transactions from remote peers, the wallet and the RPC interface.
```

**Review by jonatack — COMMENTED** [review]
ACK, LGTM apart from one comment (missing "s"). If no feedback in the next hour or two, I'll add the "s" and merge.

**Comment by jonatack** [comment]
Thanks @MarcoFalke

**Pull request merged**